src/graphql/resolvers/mutations/favoriting.js

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable import/prefer-default-export */
+import debug from 'debug';
+
+import favoriteStore from '../../../dataSources/cloudFirestore/favorite';
+import sessionStore from '../../../dataSources/cloudFirestore/session';
+
+const dlog = debug('that:api:sessions:mutation:favorite');
+
+export const fieldResolvers = {
+  FavoritingMutation: {
+    toggle: async (
+      { eventId },
+      { sessionId },
+      { dataSources: { firestore }, user },
+    ) => {
+      dlog('toggle favorite %s', sessionId);
+      // see if favorite exists
+      const fav = await favoriteStore(firestore).findSessionForMember(
+        sessionId,
+        user,
+      );
+
+      if (fav) {
+        dlog('favorite exists, removing');
+        const res = await favoriteStore(firestore).removeFavorite(fav.id);
+        dlog('favorite deleted at ', res);
+      } else {
+        dlog('favorite not exist, adding new');
+
+        const favSession = await sessionStore(firestore).findSession(sessionId);
+        if (
+          favSession &&
+          ['ACCEPTED', 'SCHEDULED', 'CANCELLED'].includes(favSession.status)
```

Comment:
this is now a check we've had multiple times.. wonder if it's just better to wrap our db call with it as the where so we don't have to worry about it everywhere

Reply:
@csell5 The database is retrieving the document by Id, from what I am finding you can't request the document by id AND put a where condition on it to only return if x is true. I also see no query for `docid == x && blah == y`